### ISSUE TITLE
--cuda option with strange results

### ISSUE THREAD

**Issue by fmobrj** [SeanNaren/deepspeech.pytorch#356]
Hello. Thanks for the great work. 

When I try to train withoy using my GPU, without the --cuda option set, the training seems to go on normally, with the loss slowly droping, as I would expect. As expected, no GPU activity, using nvidia-smi.

$ python train.py --train-manifest data/libri_train_manifest.csv --val-manifest data/libri_val_manifest.csv

Without the cuda option training results:

![image](https://user-images.githubusercontent.com/22793365/50664174-55774600-0f93-11e9-99e5-8ff0e5475686.png)

When I use the --cuda option, the nvidia-smi shows the GPU usage and the training advances much faster, but the results are somewhat strange. The loss starts ate 0 and stays like that for all the training:

![image](https://user-images.githubusercontent.com/22793365/50664290-b1da6580-0f93-11e9-999c-d15b8cfcfbcb.png)

It seems to be some kind of problem with the input tensor. Any hints?

Best regards,
Fabio.

**Comment by yuxwang1102** [comment]
Related issue. Neither my cpu or gpu works with deepspeech. The test for warpctc works fine, 
 https://github.com/SeanNaren/warp-ctc/issues/105

**Comment by yuxwang1102** [comment]
Hi fmobrj,
    Thanks a lot for your help.
    Could you please tell me your GCC and torch version? I tried the way you suggested with bad luck. The losses remain 0.

Best,

**Comment by fmobrj** [comment]
You are welcome, my friend.

>>> import torch
>>> torch.__version__
'1.0.0'
>>> torch.version.cuda
'9.0.176'

gcc version 6.5.0 20181026 (Ubuntu 6.5.0-2ubuntu1~16.04)

**Comment by miguelvr** [comment]
hey guys, AFAIK both repos work with pytorch 0.4.X and weren't really tested with pytorch 1.0

**Comment by SeanNaren** [comment]
@miguelvr you think that is the issue? sorry for the silence, I'll hopefully get time to address this ASAP

**Comment by fmobrj** [comment]
Thanks, Miguel!

For some reason, I couldn't make it work with --cuda be it torch 0.4.X or 1.0.0 using the warp-ctc binding from the README instructions. After trying everything (updated gcc version, multiple torch versions, even Python versions), I found the alternative repo for the ctc bindings and tried with 1.0.0 and it is working for me. 

I suspect it has something to do with the cuda driver.  

Hope yux can figure out an environment that works for him.

**Comment by miguelvr** [comment]
@SeanNaren no clue, but it would be worth a try. Also, @fmobrj seems to be using windows, so there's that.

**Comment by fmobrj** [comment]
> @SeanNaren no clue, but it would be worth a try. Also, @fmobrj seems to be using windows, so there's that.

Miguel, I am using Ubuntu, 18.04.1.

**Comment by fmobrj** [comment]
@miguelvr I am using Ubuntu, 18.04.1.

**Comment by miguelvr** [comment]
oh sorry, my bad.

**Comment by fmobrj** [comment]
No problem.

**Comment by fmobrj** [comment]
> @miguelvr you think that is the issue? sorry for the silence, I'll hopefully get time to address this ASAP

Thank you very much, @SeanNaren.

**Comment by SeanNaren** [comment]
Could you change `sound.shape[1] == 1:` to `sound.shape[0] == 1:`  at: https://github.com/SeanNaren/deepspeech.pytorch/blob/master/data/data_loader.py#L26 and tell me if it works?

**Comment by yuxwang1102** [comment]
@SeanNaren  @fmobrj I got normal loss now. Thanks a lot for your help！
Here I list something for those who need it.
torch version: 0.4.1
gcc version: 4.8.4
ubuntu version: 14.04
for the lack of extension.h trouble, refer to 
https://github.com/pytorch/audio/issues/71
https://github.com/SeanNaren/warp-ctc/issues/101
 
If your warp-ctc works fine but your loss remains 0 whether --cuda or not, do as what @SeanNaren says.

**Comment by SeanNaren** [comment]
Well thats a mess... I'll investigate further for whatever happened, it seems like the dimensions have swapped from `torchaudio` so will just have to transpose before we do any other transformations. Will push a fix once i verify it works!

**Comment by SeanNaren** [comment]
Fixed on the master branch!
